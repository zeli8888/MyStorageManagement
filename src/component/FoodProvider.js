import React, { Component, useState, useEffect } from 'react';
import IngredientService from '../service/IngredientService'
import DishService from '../service/DishService'

export const FoodContext = React.createContext();  //exporting context object

const FoodProvider = ({ children }) => {
    const [ingredients, setIngredients] = useState([]);
    const [dishes, setDishes] = useState([]);

    useEffect(() => {
        IngredientService.getAllIngredients().then(response => { setIngredients(response.data) })
        DishService.getAllDishes().then(response => { setDishes(response.data) })
    }, []);

    return (
        <FoodContext.Provider value={
            {
                ingredients: ingredients,
                setIngredients: setIngredients,
                dishes: dishes,
                setDishes: setDishes
            }}>
            {children
                //this indicates that all the child tags with MyProvider as Parent can access the global store  
            }
        </FoodContext.Provider>)
}

export default FoodProvider;