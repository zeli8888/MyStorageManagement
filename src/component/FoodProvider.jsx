import React, { useState, useEffect } from 'react';
import ingredientService from '../service/IngredientService'
import dishService from '../service/DishService'

export const FoodContext = React.createContext();  //exporting context object

const FoodProvider = ({ children }) => {
    const [allIngredients, setAllIngredients] = useState([]);
    const [allDishes, setAllDishes] = useState([]);

    useEffect(() => {
        ingredientService.getAllIngredients().then(response => { setAllIngredients(response.data) }).catch(error => { console.log(error) });
        dishService.getAllDishes().then(response => { setAllDishes(response.data) }).catch(error => { console.log(error) });
    }, []);

    return (
        <FoodContext.Provider value={
            {
                allIngredients: allIngredients,
                setAllIngredients: setAllIngredients,
                allDishes: allDishes,
                setAllDishes: setAllDishes
            }}>
            {children
                //this indicates that all the child tags with MyProvider as Parent can access the global store  
            }
        </FoodContext.Provider>)
}

export default FoodProvider;