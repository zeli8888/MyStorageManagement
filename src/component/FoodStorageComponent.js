import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { DataGrid } from '@mui/x-data-grid';
import IngredientService from '../service/IngredientService'

const FoodStorageComponent = function () {
    const navigate = useNavigate();

    const [ingredients, setIngredients] = useState([]);

    useEffect(() => {
        refreshIngredients();
    }, [navigate]);

    const refreshIngredients = () => {
        IngredientService.getAllIngredients().then(response => {
            setIngredients(response.data);
        }).catch(error => {
            console.log(error);
        });
    }

    const columns = [
        {
            field: 'ingredientName',
            headerName: 'Ingredient',
            width: 150,
            editable: true,
        },
        {
            field: 'ingredientStorage',
            headerName: 'Storage',
            type: 'number',
            width: 90,
            editable: true,
        },
        {
            field: 'ingredientCost',
            headerName: 'Cost',
            type: 'number',
            width: 90,
            editable: true,
        },
        {
            field: 'ingredientDesc',
            headerName: 'Description',
            width: 110,
            editable: true,
        }
    ];

    // const rows = [
    //     { ingredient: 'Beef', storage: 2, cost: 6, description: '610g/6â‚¬' },
    // ];

    return (
        <div className="container">
            <DataGrid
                rows={ingredients}
                columns={columns}
                initialState={{
                    pagination: {
                        paginationModel: {
                            pageSize: 10,
                        },
                    },
                }}
                pageSizeOptions={[10, 20]}
                checkboxSelection
                disableRowSelectionOnClick
                getRowId={(row) => row.ingredientName}
            />
        </div>
    )
}

export default FoodStorageComponent;