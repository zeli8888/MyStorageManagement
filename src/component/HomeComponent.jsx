import { useEffect, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import dishRecordService from '../service/DishRecordService';
import { PieChartCard } from './utils'
import Grid from '@mui/material/Grid';
import { DataGrid } from '@mui/x-data-grid';
const columns = [
    {
        field: 'ingredientId',
        headerName: 'Id',
        width: 0,
        editable: false,
        headerAlign: 'center'
    },
    {
        field: 'ingredientName',
        headerName: 'Ingredient',
        flex: 2,
        editable: false,
        headerAlign: 'center'
    },
    {
        field: 'supplyDays',
        headerName: 'Supply Days',
        flex: 2,
        type: 'number',
        editable: false,
        headerAlign: 'center'
    },
    {
        field: 'dailyUsage',
        headerName: 'Daily Usage',
        flex: 2,
        type: 'number',
        editable: false,
        headerAlign: 'center'
    },
    {
        field: 'totalUsage',
        headerName: 'Total Usage',
        flex: 2,
        type: 'number',
        editable: false,
        headerAlign: 'center'
    },
    {
        field: 'totalCost',
        headerName: 'Total Cost',
        flex: 2,
        type: 'number',
        editable: false,
        headerAlign: 'center'
    },
    {
        field: 'ingredientStorage',
        headerName: 'Storage',
        flex: 2,
        type: 'number',
        editable: false,
        headerAlign: 'center'
    },
    {
        field: 'ingredientCost',
        headerName: 'Cost',
        flex: 2,
        type: 'number',
        editable: false,
        headerAlign: 'center'
    },
    {
        field: 'ingredientDesc',
        headerName: 'Description',
        flex: 3,
        editable: false,
        headerAlign: 'center'
    }
];

export default function HomeComponent() {
    const navigate = useNavigate();
    const [foodRecordAnalysis, setFoodRecordAnalysis] = useState();
    const [endTime, setEndTime] = useState(new Date().toISOString());
    const [startTime, setStartTime] = useState(() => {
        const date = new Date();
        date.setMonth(date.getMonth() - 1);
        return date.toISOString();
    });

    const refresh = () => {
        dishRecordService.getDishRecordAnalysis(startTime, endTime).then(response => {
            setFoodRecordAnalysis(response.data);
            console.log(response.data);
        }).catch(error => {
            console.log(error);
        });
    }

    useEffect(() => {
        refresh();
    }, [navigate]);

    return (
        foodRecordAnalysis && <div>
            <Grid container spacing={2} direction="row" width={'100%'}>
                <Grid size={{ md: 12, lg: 10 }} width={'100%'}>
                    <DataGrid
                        rows={foodRecordAnalysis.ingredientsSummary.map(
                            summary => ({
                                ...summary.ingredient,
                                ...summary,
                            })
                        )}
                        columns={columns}
                        initialState={{
                            sorting: {
                                sortModel: [{
                                    field: 'supplyDays',
                                    sort: 'asc',
                                }]
                            },
                            columns: {
                                columnVisibilityModel: {
                                    ingredientId: false,
                                    ingredientCost: false,
                                    ingredientDesc: false,
                                    ingredientStorage: false,
                                }
                            },
                            pagination: {
                                paginationModel: {
                                    pageSize: 10,
                                }
                            }
                        }}
                        pageSizeOptions={[5, 10, 15, 20, 25]}
                        getRowId={(row) => row.ingredientId}
                        disableRowSelectionOnClick
                        sx={{
                            boxShadow: 2,
                            border: 2,
                            borderColor: 'lightgrey',
                            '& .MuiDataGrid-cell': {
                                textAlign: 'center',
                            }
                        }}
                    />
                </Grid>

                <Grid size={{ md: 12, lg: 2 }} width={'100%'}>
                    <Grid container direction="column" spacing={1}>
                        <Grid sx={{ flex: 1 }}>
                            <PieChartCard
                                data={foodRecordAnalysis.ingredientsSummary.map(
                                    summary => ({
                                        id: summary.ingredient.ingredientId,
                                        label: summary.ingredient.ingredientName,
                                        value: summary.totalCost
                                    }))
                                }
                                title="Ingredients Cost"
                                showLinear={false}
                            />
                        </Grid>

                        <Grid sx={{ flex: 5 }}>
                            <PieChartCard
                                data={foodRecordAnalysis.dishesSummary.map(
                                    summary => ({
                                        id: summary.dish.dishId,
                                        label: summary.dish.dishName,
                                        value: summary.totalUsage
                                    }))
                                }
                                title="Dishes Cooked"
                                showLinear={true}
                            />
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
        </div>
    )
}