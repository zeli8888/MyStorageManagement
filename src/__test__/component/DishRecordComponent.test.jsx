import { render, screen, waitFor, cleanup, fireEvent } from '@testing-library/react';
import { describe, test, vi, beforeEach, expect } from 'vitest';
import DishRecordComponent from '../../component/DishRecordComponent';
import { FoodContext } from '../../component/FoodProvider';
import dishRecordService from '../../service/DishRecordService';
import '@testing-library/jest-dom/vitest';

// Mock navigation hook
vi.mock('react-router-dom', () => ({
    useNavigate: () => vi.fn(),
}));

// Mock service layer
vi.mock('../../service/DishRecordService', () => ({
    default: {
        searchDishRecords: vi.fn().mockResolvedValue({
            data: {
                content: [],
                totalElements: 0
            }
        }),
        updateDishRecord: vi.fn().mockResolvedValue({}),
        deleteDishRecords: vi.fn().mockResolvedValue({}),
    },
}));

// Mock complex child components
vi.mock('../../component/utils', async (importOriginal) => ({
    ...await importOriginal(),
    EnhancedTableToolbar: ({ tableTitle, onSearch, deleteSelected, updateSelected }) => (
        <div>
            {tableTitle}
            <input data-testid="search-input" onChange={(e) => onSearch(e.target.value)} />
            <button data-testid="delete-button" onClick={deleteSelected}>Delete</button>
            <button data-testid="update-button" onClick={updateSelected}>Update</button>
        </div>
    ),
    DeletionConfirmationComponent: ({ open, onClose, onConfirm, warningMessage }) => (
        <div>
            {open && (
                <div>
                    <p>{warningMessage}</p>
                    <button onClick={onConfirm}>Confirm</button>
                    <button onClick={onClose}>Cancel</button>
                </div>
            )}
        </div>
    ),
    EnhancedTableHead: () => <thead data-testid="enhanced-table-head" />,
}));

describe('DishRecord Component Integration Tests', () => {
    const mockDishRecords = [
        {
            dishRecordId: 1,
            dishRecordTime: '2023-01-01T12:00:00',
            dishRecordDesc: 'Test Record',
            dish: {
                dishId: 1,
                dishName: 'Test Dish'
            },
            dishRecordIngredients: [
                {
                    dishRecordIngredientQuantity: 2,
                    ingredient: {
                        ingredientId: 1,
                        ingredientName: 'Test Ingredient',
                        ingredientStorage: 5,
                        ingredientCost: 10,
                        ingredientDesc: 'Test Ingredient Desc',
                    }
                }
            ]
        }
    ];

    const mockContextValue = {
        allIngredients: [
            {
                ingredientId: 1,
                ingredientName: 'Test Ingredient',
                ingredientStorage: 5,
                ingredientCost: 10,
                ingredientDesc: 'Test Ingredient Desc',
            }
        ],
        allDishes: [
            {
                dishId: 1,
                dishName: 'Test Dish',
                dishDesc: 'Test Dish Desc'
            }
        ]
    };

    beforeEach(() => {
        vi.clearAllMocks();
        dishRecordService.searchDishRecords.mockResolvedValue({
            data: {
                content: mockDishRecords,
                totalElements: 1
            }
        });
    });
    afterEach(cleanup);

    // Base snapshot verification
    test('initial render matches snapshot', async () => {
        const { asFragment } = render(
            <FoodContext.Provider value={mockContextValue}>
                <DishRecordComponent />
            </FoodContext.Provider>
        );
        await waitFor(() => expect(screen.getByText('Test Record')).toBeInTheDocument());
        expect(asFragment()).toMatchSnapshot();
    });

    // Search functionality
    test('search triggers API call', async () => {
        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishRecordComponent />
            </FoodContext.Provider>
        );
        await waitFor(() => screen.getByTestId('search-input'));
        fireEvent.change(screen.getByTestId('search-input'), { target: { value: 'test' } });
        await waitFor(() => {
            expect(dishRecordService.searchDishRecords).toHaveBeenCalledWith('test', 0, 10);
        });
    });

    // Update functionality
    test('existing record update', async () => {
        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishRecordComponent />
            </FoodContext.Provider>
        );
        await waitFor(() => screen.getByText('Test Record'));
        const checkboxes = screen.getAllByRole('checkbox');
        fireEvent.click(checkboxes[1]); // Select first record
        fireEvent.click(screen.getByTestId('update-button'));

        fireEvent.change(screen.getByLabelText(/Record Description/i), {
            target: { value: 'Updated Description' }
        });
        fireEvent.click(screen.getByText('Save'));

        await waitFor(() => {
            expect(dishRecordService.updateDishRecord).toHaveBeenCalledWith(
                1,
                expect.objectContaining({
                    dishRecord: expect.objectContaining({
                        dishRecordDesc: "Updated Description"
                    })
                })
            );
        });
    });

    // Delete functionality
    test('deletes selected records', async () => {
        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishRecordComponent />
            </FoodContext.Provider>
        );
        await waitFor(() => screen.getByText('Test Record'));
        const checkboxes = screen.getAllByRole('checkbox');
        fireEvent.click(checkboxes[1]); // Select first record
        fireEvent.click(screen.getByTestId('delete-button'));
        fireEvent.click(screen.getByText(/confirm/i));

        await waitFor(() => {
            expect(dishRecordService.deleteDishRecords).toHaveBeenCalledWith([1]);
        });
    });

    // Error state handling
    test('error logging', async () => {
        const consoleErrorMock = vi.spyOn(console, 'log').mockImplementation(() => { });
        dishRecordService.searchDishRecords.mockRejectedValue(new Error('API Error'));

        render(
            <FoodContext.Provider value={mockContextValue}>
                <DishRecordComponent />
            </FoodContext.Provider>
        );

        await waitFor(() => {
            expect(consoleErrorMock).toHaveBeenCalledWith(new Error('API Error'));
        });
        consoleErrorMock.mockRestore();
    });

    // Expandable content verification
    test('displays ingredients when expanded', async () => {
        const { asFragment } = render(
            <FoodContext.Provider value={mockContextValue}>
                <DishRecordComponent />
            </FoodContext.Provider>
        );
        await waitFor(() => screen.getByText('Test Record'));
        fireEvent.click(screen.getByRole('button', { name: /expand row/i }));

        await waitFor(() => {
            expect(screen.getByText('Test Dish')).toBeInTheDocument();
            expect(asFragment()).toMatchSnapshot();
        });
    });
});
