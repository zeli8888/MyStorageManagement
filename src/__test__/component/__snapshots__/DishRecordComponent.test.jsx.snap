// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DishRecord Component Integration Tests > displays ingredients when expanded 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-d64g0g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      >
        <div>
          Dish Records
          <input
            data-testid="search-input"
          />
          <button
            data-testid="delete-button"
          >
            Delete
          </button>
          <button
            data-testid="update-button"
          >
            Update
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-y2ff7i-MuiPaper-root-MuiTableContainer-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <table
            aria-label="collapsible table"
            class="MuiTable-root css-4z2v8j-MuiTable-root"
          >
            <thead
              data-testid="enhanced-table-head"
            />
            <tbody
              class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
            >
              <tr
                aria-checked="false"
                class="MuiTableRow-root MuiTableRow-hover css-6ge3s-MuiTableRow-root"
                role="checkbox"
                tabindex="-1"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-mfzv8e-MuiTableCell-root"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-vuhw6e-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
                      data-indeterminate="false"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                </td>
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                  scope="row"
                >
                  Test Dish
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  Test Record
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  2023-01-01 12:00
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <button
                    aria-label="expand row"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="KeyboardArrowDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                      />
                    </svg>
                  </button>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                  style="padding-bottom: 0px; padding-top: 0px;"
                />
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiTablePagination-root css-7mt0f-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-1gak8h1-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1wtxofq-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-s09cke-MuiTablePagination-selectLabel"
              id=":r1n:"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-select MuiSelect-root MuiTablePagination-input css-oegngy-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r1n: :r1m:"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-18zjd9i-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input"
                id=":r1m:"
                role="combobox"
                tabindex="0"
              >
                10
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="10"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-ug7h84-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-11cfq65-MuiTablePagination-displayedRows"
            >
              1â€“1 of 1
            </p>
            <div
              class="MuiTablePaginationActions-root MuiTablePagination-actions css-sjsoon-MuiTablePaginationActions-root"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;

exports[`DishRecord Component Integration Tests > initial render matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-d64g0g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      >
        <div>
          Dish Records
          <input
            data-testid="search-input"
          />
          <button
            data-testid="delete-button"
          >
            Delete
          </button>
          <button
            data-testid="update-button"
          >
            Update
          </button>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-12 css-yqiqnf-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-y2ff7i-MuiPaper-root-MuiTableContainer-root"
          style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
        >
          <table
            aria-label="collapsible table"
            class="MuiTable-root css-4z2v8j-MuiTable-root"
          >
            <thead
              data-testid="enhanced-table-head"
            />
            <tbody
              class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
            >
              <tr
                aria-checked="false"
                class="MuiTableRow-root MuiTableRow-hover css-6ge3s-MuiTableRow-root"
                role="checkbox"
                tabindex="-1"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox MuiTableCell-sizeMedium css-mfzv8e-MuiTableCell-root"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-vuhw6e-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
                      data-indeterminate="false"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                </td>
                <th
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                  scope="row"
                >
                  Test Dish
                </th>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  Test Record
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  2023-01-01 12:00
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                >
                  <button
                    aria-label="expand row"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="KeyboardArrowDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                      />
                    </svg>
                  </button>
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
                  colspan="5"
                  style="padding-bottom: 0px; padding-top: 0px;"
                />
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiTablePagination-root css-7mt0f-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-1gak8h1-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1wtxofq-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-s09cke-MuiTablePagination-selectLabel"
              id=":r1:"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-select MuiSelect-root MuiTablePagination-input css-oegngy-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r1: :r0:"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-18zjd9i-MuiNativeSelect-root-MuiSelect-select-MuiInputBase-input"
                id=":r0:"
                role="combobox"
                tabindex="0"
              >
                10
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
                tabindex="-1"
                value="10"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-ug7h84-MuiSvgIcon-root-MuiNativeSelect-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-11cfq65-MuiTablePagination-displayedRows"
            >
              1â€“1 of 1
            </p>
            <div
              class="MuiTablePaginationActions-root MuiTablePagination-actions css-sjsoon-MuiTablePaginationActions-root"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-h7cjts-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
